types/index.d.ts

```diff
@@ -125,3 +125,19 @@ export type UnwrapPromise<T> = T extends Promise<infer U>
 	: T extends (...args: any) => infer U
 	? U
 	: T;
```

Comment:
keen to learn whats going on here ^ looks like Rust!

Reply:
yeah keen to understand this as well!

Reply:
So, for all of the async functions, the return type is always in this form `Promise<PoolExchangeInfo>`, and sometimes we want to extract the type inside the diamond brackets `<>`, and that's what that `UnwrapPromise` type does, it opens up and return the type within the `Promise`

Actually, funny enough, while writing this down, I just saw there is an official one called `Awaited` since TS `4.5`, will update the code to use that instead.

See here https://stackoverflow.com/a/49889856